### PR TITLE
odim-5840: Implement action under license service for Systems and Managers resource URLs

### PR DESCRIPTION
**Pull request by hemkapoor**
- Implement action under license service
- Add UT's

### CODE REVIEW

**Review by AngithCJ — APPROVED** [review]

File: svc-licenses/lcommon/common.go [hunk 1]
```diff
@@ -209,3 +209,16 @@ func GenericSave(body []byte, table string, key string) error {
...
 	return nil
 }
+
+func GetIDsFromURI(uri string) (string, string, error) {
+	lastChar := uri[len(uri)-1:]
+	if lastChar == "/" {
+		uri = uri[:len(uri)-1]
+	}
+	uriParts := strings.Split(uri, "/")
+	ids := strings.SplitN(uriParts[len(uriParts)-1], ".", 2)
```
Comment (shivacharanms) [inline svc-licenses/lcommon/common.go:219]: Do not use "." to split, there might be ID from a server that can have multiple "."

File: svc-licenses/licenses/licenses.go [hunk 2]
```diff
@@ -103,3 +112,144 @@ func (e *ExternalInterface) GetLicenseResource(req *licenseproto.GetLicenseResou
...
+		log.Error(errMsg)
+		return common.GeneralError(http.StatusBadRequest, response.InternalError, errMsg, nil, nil)
+	}
+	var serverURI string
+	var err error
+	var managerLink []string
+	linksMap := make(map[string]bool)
+	for _, serverIDs := range installreq.Links.Link {
+		serverURI = serverIDs.Oid
+		if strings.Contains(serverURI, "Systems") {
```
Comment (shivacharanms) [inline svc-licenses/licenses/licenses.go]: Cant, we use a switch case to future proof the code? we won't have to add else conditions every time.

**Review by shivacharanms — CHANGES_REQUESTED** [review]

File: svc-api/router/router.go [hunk 3]
```diff
@@ -684,6 +684,11 @@ func Router() *iris.Application {
 	licenseService.Get("/", licenses.GetLicenseService)
 	licenseService.Get("/Licenses", licenses.GetLicenseCollection)
 	licenseService.Get("/Licenses/{id}", licenses.GetLicenseResource)
+	licenseService.Post("/Licenses", licenses.InstallLicenseService)
+	licenseService.Any("/", handle.LicenseMethodNotAllowed)
+	licenseService.Any("/Licenses", handle.LicenseMethodNotAllowed)
+	licenseService.Any("/Licenses/{id}", handle.LicenseMethodNotAllowed)
+	licenseService.Any("/Licenses", handle.LicenseMethodNotAllowed)
```
Comment (amar-shalgar) [inline svc-api/router/router.go]: this is duplicate

File: svc-licenses/licenses/licenses_test.go [hunk 4]
```diff
@@ -92,3 +92,87 @@ func TestGetLicenseResource(t *testing.T) {
 	assert.Equal(t, int(response.StatusCode), http.StatusOK, "Status code should be StatusOK.")
 	assert.Equal(t, licenseResourceResponse, license, "Valid License resource response is expected")
 }
+
+func TestInstallLicenseService(t *testing.T) {
+	req := &licenseproto.InstallLicenseRequest{
+		RequestBody: []byte(`{
+			"LicenseString": "333K2-8QLV4-Y8R3G-LJQQX-7BK6M",
```
Comment (amar-shalgar) [inline svc-licenses/licenses/licenses_test.go]: is this the dummy license string?

**Review by amar-shalgar — CHANGES_REQUESTED** [review]

File: svc-licenses/licenses/licenses_test.go (hunk 4 above)
Comment (shivacharanms) [reply svc-licenses/licenses/licenses_test.go]: Change it to "XXXX-XXXX-XXXX-XXXX-XXXX"

Comment (hemkapoor) [reply svc-licenses/licenses/licenses_test.go]: As of now i have kept string as XYZ, will change it to "XXXX-XXXX-XXXX-XXXX-XXXX" in my next PR for aggregates.

**Review by amar-shalgar — APPROVED** [review]

**Review by shivacharanms — APPROVED** [review]

**Pull request merged**